src/confluent_kafka/aio/_AIOProducer.py

```diff
@@ -21,36 +21,102 @@
+
+    async def poll(self, timeout=0, *args, **kwargs):
+        """Processes callbacks - blocking behavior depends on timeout
+        
+        Args:
+            timeout: 0 = non-blocking, >0 = block up to timeout seconds, -1 = block indefinitely
+        """
+        if timeout > 0:
+            # Blocking call - use ThreadPool to avoid blocking event loop
+            return await self._call(self._producer.poll, timeout, *args, **kwargs)
+        else:
+            # Non-blocking call (timeout=0) - direct call is safe
+            return self._producer.poll(timeout, *args, **kwargs)
+
+    # ========================================================================
+    # NON-BLOCKING OPERATIONS - Direct calls (no ThreadPool overhead)
+    # These operations are already async-safe in librdkafka
+    # ========================================================================
+
+    async def produce(self, topic, value=None, key=None, *args, **kwargs):
+        """Non-blocking: Queues message and returns immediately
+        
+        Args:
+            topic: Kafka topic name (required)
+            value: Message payload (optional)
+            key: Message key (optional)
+            *args, **kwargs: Additional parameters like partition, timestamp, headers
+        """
+        # Get current running event loop
+        result = asyncio.get_running_loop().create_future()
+        
+        # Track send time for latency calculation if metrics are enabled
+        send_time = None
+        message_id = None
+        if self._metrics_callback:
+            import time
+            send_time = time.time()
+            message_id = f"{topic}:{key}:{id(result)}"  # Unique message identifier
+            self._send_times[message_id] = send_time
+
+        # Pre-bind variables to avoid closure overhead
+        def on_delivery(err, msg):
+            if err:
+                # Handle delivery failure
+                if self._metrics_callback and message_id:
+                    latency_ms = 0.0
+                    if message_id in self._send_times:
+                        latency_ms = (time.time() - self._send_times[message_id]) * 1000
+                        del self._send_times[message_id]
+                    self._metrics_callback(latency_ms, topic, 0, False)  # success=False
+                result.set_exception(_KafkaException(err))
+            else:
+                # Handle delivery success
+                if self._metrics_callback and message_id:
+                    latency_ms = 0.0
+                    if message_id in self._send_times:
+                        latency_ms = (time.time() - self._send_times[message_id]) * 1000
+                        del self._send_times[message_id]
+                    self._metrics_callback(latency_ms, msg.topic(), msg.partition(), True)  # success=True
+                result.set_result(msg)
+        
+        kwargs['on_delivery'] = on_delivery
```

Comment:
Did we overwrite the user on_delivery argument here? I think the intermediate solution checked if the callback was async or not and called it alongside the result setting. It'd probably be good to still do that even though it's not a common pattern for async with a future returned but since we're maintaining api compatibility.